### ISSUE TITLE
Rule Change: `no-throw-literal` - option to also check promise rejections

### ISSUE THREAD

**Issue by DetachHead** [eslint/eslint#17592]
### What rule do you want to change?

no-throw-literal

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
Promise.reject('asdf')
new Promise((_, reject) => reject('asdf'))
```

### What does the rule currently do for this code?

no errors

### What will the rule do after it's changed?

error on both lines

the correct code would be:
```js
Promise.reject(new Error('asdf'))
new Promise((_, reject) => reject(new Error('asdf')))
```

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

a promise rejected with a non-`Error` value has the same problems as a `throw` statement with a non-`Error` value, so it should also check promise rejections

**Comment by mdjermanovic** [comment]
Hi @DetachHead, thanks for the issue!

> ```js
> Promise.reject('asdf')
> new Promise((_, reject) => reject('asdf'))
> ```

This is already covered by a separate rule [`prefer-promise-reject-errors`](https://eslint.org/docs/latest/rules/prefer-promise-reject-errors). Here's a [playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IHByZWZlci1wcm9taXNlLXJlamVjdC1lcnJvcnM6IFwiZXJyb3JcIiAqL1xuXG5cbi8vIGJhZFxuUHJvbWlzZS5yZWplY3QoJ2FzZGYnKVxubmV3IFByb21pc2UoKF8sIHJlamVjdCkgPT4gcmVqZWN0KCdhc2RmJykpXG5cbi8vIGdvb2RcblByb21pc2UucmVqZWN0KG5ldyBFcnJvcignYXNkZicpKVxubmV3IFByb21pc2UoKF8sIHJlamVjdCkgPT4gcmVqZWN0KG5ldyBFcnJvcignYXNkZicpKSkiLCJvcHRpb25zIjp7ImVudiI6eyJlczYiOnRydWV9LCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e30sImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCJ9LCJydWxlcyI6e319fQ==) with your code examples.
